src/PrefixTree.java

```diff
@@ -30,55 +27,51 @@ public boolean contains(CharSequence word){
     }
 
     public boolean add(CharSequence word){
-        Helper helper = traverse(word);
-        Node previousNode = helper.node;
+        TraverseResult helper = traverse(word);
+        Node currentNode = helper.node;
         if(helper.index == -1){
             System.out.println("Word has been added already. " + word);
             return false;
         }
-        else{
-            for (int i = helper.index; i < word.length(); i++) {
-                if(word.length() == 1){
-                    break;
-                }
-                Edge edge = new Edge();
-                Node node = new Node();
-                node.setValue(word.charAt(i));
-                edge.setReference(node);
-                previousNode.addEdge(edge);
-                previousNode = node;
-            }
-            previousNode.setFinal(true);
+        if(word.length() == 1){
+            currentNode.setFinal(true);
             return true;
         }
+        for (int i = helper.index; i < word.length(); i++) {
+            Node nextNode = new Node(word.charAt(i));
+            Edge edge = new Edge(nextNode);
+            currentNode.addEdge(edge);
+            currentNode = nextNode;
+            }
```

Comment:
Indentation. I'd suggest configuring auto-format on save in your IDE. It makes sense for green field development.